### PR TITLE
Implement http range requests for HttpsSeekableByteChannel

### PR DESCRIPTION
**Pull request by frcroth**
### URL of deployed dev instance (used for testing):
- https://httprangerequests.webknossos.xyz/

### Steps to test:
- View a remote dataset via HTTPS
- View a remote dataset via HTTPS with basic auth
- Run tests that actually use HTTPS range requests

### Issues:
- contributes to #6823 

------
(Please delete unneeded items, merge only when none are left open)
- I think no changelog since this is not user facing?
- [x] Considered [common edge cases](../blob/master/.github/common_edge_cases.md)
- [x] Needs datastore update after deployment

### CODE REVIEW

File: webknossos-datastore/app/com/scalableminds/webknossos/datastore/storage/httpsfilesystem/HttpsSeekableByteChannel.scala [hunk 1]
```diff
@@ -1,15 +1,14 @@
...
-
 import java.net.URI
 import java.nio.ByteBuffer
 import java.nio.channels.SeekableByteChannel
-import java.nio.charset.StandardCharsets
 import java.nio.file.OpenOption
 import java.util
-import scalaj.http.{Http, HttpResponse}
+
+// Does not work with Java 8, works with Java 11
```
Comment (fm3) [inline webknossos-datastore/app/com/scalableminds/webknossos/datastore/storage/httpsfilesystem/HttpsSeekableByteChannel.scala:9]: I’d say this comment can be removed, as you also adapted the readme (and the dockerfiles come with java 11 currently)

File: webknossos-datastore/app/com/scalableminds/webknossos/datastore/storage/httpsfilesystem/HttpsSeekableByteChannel.scala [hunk 2]
```diff
@@ -21,46 +20,85 @@ class HttpsSeekableByteChannel(path: HttpsPath, openOptions: util.Set[_ <: OpenO
...
+  }.getOrElse(
+      basicRequest
+    )
+    .readTimeout(readTimeout)
+
+  private lazy val headRequest: Request[Either[String, String], Any] =
+    authenticatedRequest.head(Uri(uri))
+
+  private def getHeaderInformation = {
+    val response: Identity[Response[Either[String, String]]] = backend.send(headRequest)
```
Comment (fm3) [inline webknossos-datastore/app/com/scalableminds/webknossos/datastore/storage/httpsfilesystem/HttpsSeekableByteChannel.scala:39]: Do you know what’s the deal with the Either[String,String] here? Did you chose this or did it come from the library? What’s the semantics?

**Review by fm3 — COMMENTED** [review]
Looking good! Could you add an entry to the migration guide, mentioning that java 8 is no longer supported?
The wk code does not currently use the range requests, right? So the steps to test is just to ensure that the read fully codepath still works?

File: webknossos-datastore/app/com/scalableminds/webknossos/datastore/storage/httpsfilesystem/HttpsSeekableByteChannel.scala (hunk 2 above)
Comment (frcroth) [reply webknossos-datastore/app/com/scalableminds/webknossos/datastore/storage/httpsfilesystem/HttpsSeekableByteChannel.scala:38]: It comes from the library, where the left side of the either is for error message and the right side for successful responses

Comment (fm3) [reply webknossos-datastore/app/com/scalableminds/webknossos/datastore/storage/httpsfilesystem/HttpsSeekableByteChannel.scala:38]: I see. Well, fair enough, thanks!

**Review by fm3 — APPROVED** [review]

**Pull request merged**